aptos-move/framework/aptos-framework/sources/Transaction.move

```diff
@@ -0,0 +1,28 @@
+/// Native Move functions for transaction metadata and scheduling
+module AptosFramework::Transaction {
+
+    use AptosFramework::TypeInfo
+    use Std::ASCII::String;
+
+    /// Return the version number of the distributed database,
+    /// corresponding to the number of transactions the system will have
+    /// executed upon the conclusion of the current one. Reference:
+    /// [Versioned database](https://aptos.dev/basics/basics-txns-states#versioned-database)
+    public native fun version_number(): u64;
+
+    /// Schedule transaction for the epilogue of a block, delaying by
+    /// `delay` blocks. A `delay` of 0 schedules a transaction during
+    /// the epilogue of the current block, and can only be called from a
+    /// non-epilogue transaction. Calling during the epilogue requires
+    /// a `delay` of at least 1, corresponding to a transaction
+    /// scheduled during the epilogue of the next block.
+    public(script) native fun schedule(
+        account: &signer,
+        script_address: address,
+        script_module_name: String,
+        script_function_name: String,
+        type_arguments: &vector<TypeInfo>,
+        arguments: &vector<String>,
+        delay: u64,
+    );
+}
```

Comment:
@msmouse How about this? Based on past discussion with the core dev team about an event scheduler, I'm assuming that scheduled transactions would be executed during what I'm calling the "block epilogue"

Reply:
This seems overall okay as a concept, do you want to build out the idea?

Reply:
@davidiw Sounds like a plan. Right now the order book's atomic matching engine is being implemented, and when that is complete next up is parallelized order queueing (which leverages the version number native function in this proposal), and automated batchwise order matching on a regular interval (which leverages the event scheduler in this proposal). It's looking like the proposed functionality may require integration with the [parallel executor scheduler](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/parallel-executor/src/executor.rs) and the Move VM, and I look forward to discussing this further when the incubation program starts and in-person collaboration is possible